vars/sectionBuildAndTest.groovy

```diff
@@ -90,13 +90,26 @@ def call(params) {
             withAcrClient(subscription) {
 
               def acbTemplateFilePath = 'acb.tpl.yaml'
+              def dockerfileTest = 'Dockerfile_test'
+              def isOnMaster = new ProjectBranch(env.BRANCH_NAME).isMaster()
 
               pcr.callAround('dockerbuild') {
                 timeoutWithMsg(time: 15, unit: 'MINUTES', action: 'Docker build') {
                   def buildArgs = projectBranch.isPR() ? " --build-arg DEV_MODE=true" : ""
-                  fileExists(acbTemplateFilePath) ?
+                  if (fileExists(acbTemplateFilePath)) {
                     acr.runWithTemplate(acbTemplateFilePath, dockerImage)
-                    : acr.build(dockerImage, buildArgs)
+                  } else {
+                    acr.build(dockerImage, buildArgs)
+                  }
+                  if (isOnMaster && fileExists('build.gradle')) {
```

Comment:
I think this either needs more testing with different projects, i.e. a couple on java 8 at least or a feature toggle, we could use a feature toggle while we roll it out on our apps and if we're happy with the feature just remove the toggle and enable it for everyone?

Reply:
Either feature toggle or we change the projects builds. Maybe the latter is actually easier.

Reply:
🤷‍♂ needs testing before rolling something like this out to everyone though

Reply:
It's safer to point some projects to this branch.